llvm/lib/Target/RISCV/RISCVTargetMachine.cpp

```diff
@@ -389,6 +390,13 @@ class RISCVPassConfig : public TargetPassConfig {
       DAG->addMutation(createStoreClusterDAGMutation(
           DAG->TII, DAG->TRI, /*ReorderWhileClustering=*/true));
     }
+
+    const RISCVSubtarget &ST = C->MF->getSubtarget<RISCVSubtarget>();
+    if (!ST.getMacroFusions().empty() && ST.useLoadStorePairs()) {
```

Comment:
This needs to be integrated with the `EnablePostMISchedLoadStoreClustering` on line 386. Otherwise we will have multiple load/store cluster mutations.